tox.ini

```diff
@@ -105,56 +95,24 @@ commands =
 [testenv:lint]
 basepython: python3.7
 deps =
-  pylint~=2.3
-  flake8~=3.7
-  isort~=4.3
-  black>=19.3b0,==19.*
+  -c dev-requirements.txt
+  pylint
+  flake8
+  isort
+  black
 
 commands_pre =
-  pip install -e {toxinidir}/opentelemetry-api
-  pip install -e {toxinidir}/opentelemetry-sdk
-  pip install -e {toxinidir}/ext/opentelemetry-ext-http-requests
-  pip install -e {toxinidir}/ext/opentelemetry-ext-jaeger
-  pip install -e {toxinidir}/ext/opentelemetry-ext-pymongo
-  pip install -e {toxinidir}/ext/opentelemetry-ext-testutil
-  pip install -e {toxinidir}/ext/opentelemetry-ext-wsgi
-  pip install -e {toxinidir}/ext/opentelemetry-ext-flask[test]
-  pip install -e {toxinidir}/examples/opentelemetry-example-app
-  pip install -e {toxinidir}/ext/opentelemetry-ext-opentracing-shim
+  python scripts/eachdist.py install --editable
 
 commands =
-; Prefer putting everything in one pylint command to profit from duplication
-; warnings.
-  black --check --diff .
-  pylint opentelemetry-api/src/opentelemetry \
-         opentelemetry-api/tests/ \
-         opentelemetry-sdk/src/opentelemetry \
-         opentelemetry-sdk/tests/ \
-         ext/opentelemetry-ext-http-requests/src/ \
-         ext/opentelemetry-ext-http-requests/tests/ \
-         ext/opentelemetry-ext-jaeger/src/opentelemetry \
-         ext/opentelemetry-ext-jaeger/tests/ \
-         ext/opentelemetry-ext-opentracing-shim/src/ \
-         ext/opentelemetry-ext-opentracing-shim/tests/ \
-         ext/opentelemetry-ext-pymongo/src/opentelemetry  \
-         ext/opentelemetry-ext-pymongo/tests/ \
-         ext/opentelemetry-ext-testutil/src/opentelemetry \
-         ext/opentelemetry-ext-wsgi/src/ \
-         ext/opentelemetry-ext-wsgi/tests/ \
-         ext/opentelemetry-ext-flask/src/ \
-         ext/opentelemetry-ext-flask/tests/ \
-         examples/opentelemetry-example-app/src/opentelemetry_example_app/ \
-         examples/opentelemetry-example-app/tests/ \
-         examples/basic_tracer/ \
-         examples/http/
-  flake8 .
-  isort --check-only --diff --recursive .
+  python scripts/eachdist.py lint --check-only
 
 [testenv:docs]
 deps =
-  sphinx~=2.1
-  sphinx-rtd-theme~=0.4
-  sphinx-autodoc-typehints~=1.10.2
+  -c dev-requirements.txt
+  sphinx
+  sphinx-rtd-theme
+  sphinx-autodoc-typehints
   opentracing~=2.2.0
   Deprecated>=1.2.6
```

Comment:
How come no `-c dev-requirements.txt` for `testenv:py37-tracecontext`?

Reply:
It seems that environment doesn't use any requirement that is listed in dev-requirements. Also, because the dev-requirements file is used for installing too (https://github.com/open-telemetry/opentelemetry-python/pull/291#discussion_r354190296), I didn't want to include all that.